src/canister/user_index/src/api/user_record/get_user_canister_list.rs

```diff
@@ -16,6 +16,18 @@ fn get_user_canister_list() -> Vec<Principal> {
     })
 }
 
+#[query(guard = "is_reclaim_canister_id")]
+fn get_user_id_and_canister_list() -> Vec<(Principal, Principal)> {
+    CANISTER_DATA.with(|canister_data_ref_cell| {
+        canister_data_ref_cell
+            .borrow()
+            .user_principal_id_to_canister_id_map
+            .iter()
+            .map(|(principal_id, canister_id)| (*principal_id, *canister_id))
+            .collect()
+    })
+}
```

Comment:
Why is this required?

Reply:
Offchain agent gets a list of (user_id,canister_id). Filters acc to last_canister_access_time, fires call to user index to recycle and calls yral-metadata with user_id to remove keys

Reply:
okay so do we still require `get_user_canister_list`

Reply:
Yeah, being used for getting list of canisters for metrics collection